src/runtime/parser/index.ts

```diff
@@ -97,7 +97,8 @@ export const createMarkdownParser = async (inlineOptions: MDCParseOptions = {})
     const { content, data: frontmatter } = await parseFrontMatter(md)
 
     // Start processing stream
-    const processedFile = await processor.process({ value: content, data: frontmatter })
+    const mdcWithContext = { ...(inlineOptions.rootDocument || {}), value: content, data: frontmatter }
+    const processedFile = await processor.process(mdcWithContext)
```

Comment:
VFile is directly related to the current processing task. Passing a predefined `rootDocument` for all processing tasks may cause invalid results. Especially with Nuxt Content's multi-source feature.

Instead, we should define a second argument for  `return async (md: string): Promise<MDCParserResult>` function to receive `VFile` data.

```ts
  return async (md: string, vFileData: VFile = {}): Promise<MDCParserResult> => {
    // Extract front matter data
    const { content, data: frontmatter } = await parseFrontMatter(md)

    // Start processing stream
    const processedFile = await processor.process({ ...vFileData, value: content, data: frontmatter })
    ...
  }
```